tsdb/engine/tsm1/engine.go

```diff
@@ -700,111 +698,87 @@ func (e *Engine) Export(w io.Writer, basePath string, start time.Time, end time.
 			tombstonePath = filepath.Base(r.TombstoneFiles()[0].Path)
 		}
 
-		// Skip this file if no overlap with start/end
 		min, max := r.TimeRange()
-		if min > end.UnixNano() || max < start.UnixNano() {
-			r.Close()
-			continue
-		}
 
-		r.Close()
+		// We overlap time ranges, we need to filter the file
+		if min >= start.UnixNano() && min <= end.UnixNano() ||
+			max >= start.UnixNano() && max <= end.UnixNano() {
+			if err := e.filterFileToGzip(r, file, basePath, filepath.Join(path, file), start.UnixNano(), end.UnixNano(), zw); err != nil {
+				return err
+			}
 
-		if tombstonePath != "" {
-			whole = append(filtered, tombstonePath)
 		}
 
-		if min > start.UnixNano() && max < end.UnixNano() {
-			whole = append(filtered, file)
-		} else {
-			// at this point, we know we're not totally outside or inside start and end so we'll need to scan the blocks
-			filtered = append(filtered, file)
-		}
-	}
+		// above is the only case where we need to keep the reader open.
+		r.Close()
```

Comment:
This looks like it should be within the loop above or every reader will leak except the last one.

Reply:
Also, error is ignored here.